daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala

```diff
@@ -2205,6 +2215,31 @@ case class DFDLXBitExpr(nameAsParsed: String, fnQName: RefQName,
     res
   }
 }
+
+case class DFDLXBitUnaryExpr(nameAsParsed: String, fnQName: RefQName,
+                        args: List[Expression], constructor: (CompiledDPath, NodeInfo.Kind) => RecipeOp)
+  extends FunctionCallBase(nameAsParsed, fnQName, args) {
+  override lazy val inherentType = {
+    val arg0Type = args(0).inherentType
+    schemaDefinitionUnless(
+      arg0Type.isSubtypeOf(NodeInfo.PrimType.UnsignedLong) || arg0Type.isSubtypeOf(NodeInfo.PrimType.Long),
+      "Both arguments for %s must be either xs:unsignedLong or xs:long or a subtype of those, but was %s.",
+      nameAsParsed,arg0Type.globalQName)
```

Comment:
Steve's comment about `Both arguments ...` above still needs to be addressed.  Also, we prefer spaces between comma-separated arguments on this line (2227) and line 2237 (also compare lines 2236 & 2237 with line 2253 & 2254 - please follow the same formatting as existing code).